qiskit/providers/aer/pulse_new/models/quantum_models.py

```diff
@@ -0,0 +1,245 @@
+from .operator_models import OperatorModel
+from ..type_utils import vec_commutator, vec_dissipator
+
+class HamiltonianModel(OperatorModel):
+    """A model of a Hamiltonian, i.e. a time-dependent operator of the form
+
+    .. math::
+
+        H(t) = \sum_{i=0}^{k-1} s_i(t) H_i,
+
+    where :math:`H_i` are Hermitian operators, and the :math:`s_i(t)` are
+    time-dependent functions represented by :class:`Signal` objects.
+
+    Currently the functionality of this class is as a subclass of
+    :class:`OperatorModel`, with the following modifications:
+        - The operators in the linear decomposition are verified to be
+          Hermitian.
+        - Frames are dealt with assuming the structure of the Schrodinger
+          equation. I.e. Evaluating the Hamiltonian :math:`H(t)` in a
+          frame :math:`F = -iH`, evaluates the expression
+          :math:`e^{-tF}H(t)e^{tF} - H`. This is in contrast to
+          the base class :class:`OperatorModel`, which would ordinarily
+          evaluate :math:`e^{-tF}H(t)e^{tF} - F`.
+    """
+
+    def __init__(self,
+                 operators: List[Operator],
+                 signals: Optional[Union[VectorSignal, List[Signal]]] = None,
+                 signal_mapping: Optional[Callable] = None,
+                 frame: Optional[Union[Operator, np.array]] = None,
+                 cutoff_freq: Optional[float] = None):
+        """Initialize, ensuring that the operators are Hermitian.
+
+        Args:
+            operators: list of Operator objects.
+            signals: Specifiable as either a VectorSignal, a list of
+                     Signal objects, or as the inputs to signal_mapping.
+                     OperatorModel can be instantiated without specifying
+                     signals, but it can not perform any actions without them.
+            signal_mapping: a function returning either a
+                            VectorSignal or a list of Signal objects.
+            frame: Rotating frame operator. If specified with a 1d
+                            array, it is interpreted as the diagonal of a
+                            diagonal matrix.
+            cutoff_freq: Frequency cutoff when evaluating the model.
+        """
+
+        # verify operators are Hermitian, and if so instantiate
+        for operator in operators:
+            if isinstance(operator, Operator):
+                operator = operator.data
+
+            if np.linalg.norm((operator.conj().transpose()
+                                - operator).data) > 1e-10:
+                raise Exception("""HamiltonianModel only accepts Hermitian
+                                    operators.""")
+
+        super().__init__(operators=operators,
+                         signals=signals,
+                         signal_mapping=signal_mapping,
+                         frame=frame,
+                         cutoff_freq=cutoff_freq)
+
+    def evaluate(self, time: float, in_frame_basis: bool = False) -> np.array:
```

Comment:
This is different than setting a frame. The "frame basis" throughout this code is defined as the basis in which the frame operator is diagonal (according to the output of `np.linalg.eigh`) - so setting `in_frame_basis=True` still evaluates in the frame, but returns results in basis in which the frame operator is diagonal.

This argument is purely for computational purposes - when solving in a rotating frame it makes sense to solve the problem in the basis in which the frame operator is diagonal to save on the cost of compute `expm(t * F)` for every `t` the solver queries.

I'll leave this unresolved in case you want to comment further.